crates/hir/src/lib.rs

```diff
@@ -2780,6 +2780,8 @@ impl Type {
     /// Checks that particular type `ty` implements `std::future::Future`.
     /// This function is used in `.await` syntax completion.
     pub fn impls_future(&self, db: &dyn HirDatabase) -> bool {
+        // FIXME: This should be checking for IntoFuture trait, but I don't know how to find the
+        // right TraitId in this crate.
```

Comment:
`IntoFuture::into_future` is tagged as the `into_future` lang item, so you could look for that function, then turn it into an AssocItem and fetch its trait container which should be the IntoFuture trait

Reply:
Done (and could use a review, I'm very new to this). Any suggestion for how to fix the `resolve_await_to_poll` problem?